ambry-rest/src/main/java/com.github.ambry.rest/NettyMultipartRequest.java

```diff
@@ -200,7 +200,7 @@ private void processPart(InterfaceHttpData part) throws RestServiceException {
               RestServiceErrorCode.BadRequest);
         } else {
           hasBlob = true;
-          if (fileUpload.length() != getSize()) {
+          if (getSize() != -1 && fileUpload.length() != getSize()) {
```

Comment:
We continue to let a caller specify the size to keep this class generic. However, the size is validated only if it is not -1.